enhancements/sdk-external-and-pluggable-validations.md

```diff
@@ -0,0 +1,376 @@
+* *--optional-values* would continue to be passed to the validators
+* *--output* indicates how we want to output the results.
+* *--select-optional* works as is
+* *-- alpha-select-external* added; takes in the location of the executable to run as
+  the validator, i.e. `/path/to/validator/the-executable:/path/to/another`
+
+### Risks and Mitigations
+
+There is little risk, if this does not pan out we keep going on the current path
+of compiling them in.
+
+Another possible risk is users of an airgapped environment will have to install
+these external validator executables. We could supply a mechanism for
+distributing these external validators as an image. For this first release, we
+will ignore the distribution solutions leaving it to the user to copy the
+validators onto the system running `operator-sdk`.
+
+## Design Details
+
+### Test Plan
+
+- unit tests for the feature will be added in the implementation
+- QE would need to create custom validations to be called by the new `--alpha-select-external` flag.
+
+### Graduation Criteria
+
+This feature will come out as alpha accessible via the `--alpha-select-external`
+flag to the `bundle validate` command. We will accept feedback on possible
+changes to the feature.
+
+After a period of time, we can graduate to a GA with a more permanent flag, i.e.
+`--select-external`.
+
+### Version Skew Strategy
+
+* The version of the validators can change however the validator author sees fit.
+* The API or contract between `operator-sdk` and validators will be
+  * input to validator: bundle directory
+  * output from validator: `ManifestResult` JSON
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+Validators would have to be their own executables which could result in a
+compilation step being needed depending on the language used to implement them.
+
+## Alternatives
+
+* put validations in their own images
+  * need to define "API" contract what is the entrypoint and what parameters do
+    we give it
+  * pro:
+    * already have a precendence for running things from images
+    * familiar tech
+  * con:
+    * would need a cluster to run these validations
+    * authors would have to create binaries of their validations anyway
+
+* use a language like JavaScript or CUE to define all validations
+  * validations could be run from a git repo, i.e. operator-sdk could pull it
+    and then evaluate it
+  * pro:
+    * simpler delivery, expose via a gitrepo and done
+  * con:
+    * all existing validations would have to be re-written in a new language
+      structure which could introduce new bugs
+    * unproven technology
+    * would have to write the engine to know how to execute these
+
+* use scorecard to do the validations
+  * create validations written in scorecard as custom tests
+  * pro:
+    * infrastructure required to run is already built within scorecard
+  * con:
+    * would need a cluster to run these validations
```

Comment:
```suggestion
    * would need a cluster to run these validations
    * by requiring a cluster the tests require like 3 minutes to be executed instead of milliseconds which make it very hard we are able to re-test all bundle of an index as we do via Audit Tool for example. 
```